test/robustness/overlay/areal_areal/random_integer_grids.cpp

```diff
@@ -0,0 +1,381 @@
+{
+    if(bits_in.size() == 0) return true;
+    if(bits_in.size() != (bits_size + chunk_size - 1) / chunk_size) return false;
+    if (bits_size % chunk_size != 0)
+    {
+        std::bitset<chunk_size> bm;
+        bm.set();
+        bm >>= chunk_size - bits_size % chunk_size;
+        if(bits_in.back() & ~bm.to_ullong()) return false;
+    }
+    return true;
+}
+
+int main(int argc, char** argv)
+{
+    BoostGeometryWriteTestConfiguration();
+    try
+    {
+        namespace po = boost::program_options;
+        po::options_description description("=== random_integer_grids ===\nAllowed options");
+
+        grid_settings settings;
+
+        description.add_options()
+            ("help", "Help message")
+            ("seed",
+             po::value<decltype(settings.seed)>(&settings.seed)->default_value(settings.seed),
+             "Initialization seed for random generator")
+            ("count",
+             po::value<decltype(settings.count)>(&settings.count)->default_value(settings.count),
+             "Number of tests (-1 for infinite loop)")
+            ("width",
+             po::value<decltype(settings.width)>(&settings.width)->default_value(settings.width),
+             "Width of grid (>= 1)")
+            ("height",
+             po::value<decltype(settings.height)>(&settings.height)->default_value(settings.height),
+             "Height of grid (>= 1)")
+            ("verbose",
+             po::bool_switch(&settings.verbose),
+             "Print WKT and bit patterns for each failure.")
+            ("bits1",
+             po::value<decltype(settings.bits1)>(&settings.bits1)->multitoken(),
+             "Fixed bit pattern for first operand as list of ullong.")
+            ("bits2",
+             po::value<decltype(settings.bits2)>(&settings.bits2)->multitoken(),
+             "Fixed bit pattern for second operand as list of ullong.")
+        ;
+
+        po::variables_map varmap;
+        po::store(po::parse_command_line(argc, argv, description), varmap);
+        po::notify(varmap);
+
+        if (! validate_bits_input(settings.bits1, settings.height * settings.width))
+        {
+            std::cout << "bits1 was provided but does not match dimensions.\n";
+            return 1;
+        }
+        if (! validate_bits_input(settings.bits2, settings.height * settings.width))
+        {
+            std::cout << "bits2 was provided but does not match dimensions.\n";
+            return 1;
+        }
+        if ( settings.bits1.size() != 0 && settings.bits2.size() != 0 && settings.count != 1 )
+        {
+            std::cout << "Both bit patterns fixed, count is changed to 1.\n";
+            settings.count = 1;
+        }
+        if(settings.height < 1 || settings.width < 1)
+        {
+            std::cout << "Invalid dimensions, height and width need to be positive.\n";
+            return 1;
+        }
+        if (varmap.count("help"))
```

Comment:
Nit, the layout of the `if` statements has three variants.

I would prefer this one, so: `if` `space` `(` condition `)`, which I think is the most common in our repo (though it is not always consistent)